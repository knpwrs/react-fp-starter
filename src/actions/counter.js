import R from 'ramda';
import { createAction } from 'redux-actions';

export const increment = createAction('INCREMENT', R.T);
export const decrement = createAction('DECREMENT', R.T);
export const incrementAsync = createAction('INCREMENT_ASYNC', R.T);
export const decrementAsync = createAction('DECREMENT_ASYNC', R.T);
