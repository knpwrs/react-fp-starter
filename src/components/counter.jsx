import React from 'react';
import PropTypes from 'prop-types';
import { compose, withHandlers, pure } from 'recompose';

const Counter = ({ increment, decrement, counter }) => (
  <div>
    <h1>Counter: {counter}</h1>
    <button onClick={increment}>+1</button>
    <button onClick={decrement}>-1</button>
  </div>
);

Counter.propTypes = {
  increment: PropTypes.func.isRequired,
  decrement: PropTypes.func.isRequired,
  counter: PropTypes.number.isRequired,
};

export default compose(
  pure,
  withHandlers({
    increment: props => () => props.increment(),
    decrement: props => () => props.decrement(),
  }),
)(Counter);
