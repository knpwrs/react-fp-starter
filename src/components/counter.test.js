/* eslint-disable react/jsx-filename-extension */
// https://github.com/avajs/ava/issues/631
import test from 'ava';
import R from 'ramda';
import React from 'react';
import render from 'react-test-renderer';
import { Bare as Counter } from './counter';

test('counter', (t) => {
  const tree = render.create(
    <Counter
      increment={R.T}
      decrement={R.T}
      incrementAsync={R.T}
      decrementAsync={R.T}
      counter={15}
    />,
  );
  t.snapshot(tree);
});
