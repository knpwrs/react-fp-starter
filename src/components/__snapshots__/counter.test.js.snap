// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`counter 1`] = `
<div>
  <h2>
    Counter: 
    15
  </h2>
  <button
    className="css-jltz2k"
    onClick={[Function]}
  >
    +1
  </button>
  <button
    className="css-jltz2k"
    onClick={[Function]}
  >
    +1 Aysnc
  </button>
  <button
    className="css-jltz2k"
    onClick={[Function]}
  >
    -1
  </button>
  <button
    className="css-jltz2k"
    onClick={[Function]}
  >
    -1 Async
  </button>
</div>
`;
